From 40420139292335a95ce96b69b15b561c1748aeab Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Wed, 4 Aug 2021 13:15:30 +0100
Subject: [PATCH backport 5.10 16/73] mctp: remove duplicated assignment of
 pointer hdr

The pointer hdr is being initialized and also re-assigned with the
same value from the call to function mctp_hdr. Static analysis reports
that the initializated value is unused. The second assignment is
duplicated and can be removed.

Addresses-Coverity: ("Unused value").
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 net/mctp/af_mctp.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/net/mctp/af_mctp.c b/net/mctp/af_mctp.c
index 84f722d31fd7..a9526ac29dff 100644
--- a/net/mctp/af_mctp.c
+++ b/net/mctp/af_mctp.c
@@ -170,7 +170,6 @@ static int mctp_recvmsg(struct socket *sock, struct msghdr *msg, size_t len,
 		/* TODO: expand mctp_skb_cb for header fields? */
 		struct mctp_hdr *hdr = mctp_hdr(skb);
 
-		hdr = mctp_hdr(skb);
 		addr = msg->msg_name;
 		addr->smctp_family = AF_MCTP;
 		addr->smctp_network = cb->net;
-- 
2.20.1

